libnopegl/src/node_gblur.c

```diff
@@ -0,0 +1,529 @@
+/*
+ * Copyright 2023 Matthieu Bouron <matthieu.bouron@gmail.com>
+ * Copyright 2023 Nope Forge
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <math.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "gpu_block.h"
+#include "gpu_ctx.h"
+#include "graphics_state.h"
+#include "internal.h"
+#include "log.h"
+#include "math_utils.h"
+#include "nopegl.h"
+#include "rendertarget.h"
+#include "rtt.h"
+#include "topology.h"
+#include "utils.h"
+
+/* GLSL shaders */
+#include "blur_gaussian_vert.h"
+#include "blur_gaussian_frag.h"
+
+#define _CONSTANT_TO_STR(v) #v
+#define CONSTANT_TO_STR(v) _CONSTANT_TO_STR(v)
+
+#define MAX_KERNEL_SIZE 127
+NGLI_STATIC_ASSERT(kernel_size_is_odd, MAX_KERNEL_SIZE & 0x1);
+
+#define MAX_RADIUS_SIZE 126
+NGLI_STATIC_ASSERT(radius_size, MAX_RADIUS_SIZE == (MAX_KERNEL_SIZE - 1));
+
```

Comment:
why not `#define MAX_RADIUS_SIZE (MAX_KERNEL_SIZE - 1)` then? 🤔

Reply:
Because of the macro to generate the doc string (not evaluated).